### PR TITLE
[Enhance] Add unlabeled image list to coco format scripts

### PR DESCRIPTION
**Pull request by hhaAndroid**
In order to facilitate users to test unlabeled images, a convert script has been added.

### CODE REVIEW

File: docs/1_exist_data_model.md [hunk 1]
```diff
@@ -354,7 +354,28 @@ Assume that you have already downloaded the checkpoints to the directory `checkp
 
 ### Test without Ground Truth Annotations
 
-MMDetection supports to test models without ground-truth annotations using `CocoDataset`. If your dataset format is not in COCO format, please convert them to COCO format. For example, if your dataset format is VOC, you can directly convert it to COCO format by the [script in tools.](https://github.com/open-mmlab/mmdetection/tree/master/tools/dataset_converters/pascal_voc.py)
+MMDetection supports to test models without ground-truth annotations using `CocoDataset`. If your dataset format is not in COCO format, please convert them to COCO format. For example, if your dataset format is VOC, you can directly convert it to COCO format by the [script in tools.](https://github.com/open-mmlab/mmdetection/tree/master/tools/dataset_converters/pascal_voc.py) if your dataset format is VOC, you can directly convert it to COCO format by the [script in tools.](https://github.com/open-mmlab/mmdetection/tree/master/tools/dataset_converters/cityscapes.py) The rest of the formats can be converted using [this script](https://github.com/open-mmlab/mmdetection/tree/master/tools/dataset_converters/images2coco.py).
```
Comment (RangiLyu) [inline docs/1_exist_data_model.md:357]: Duplicated `if your dataset format is VOC...`

File: docs/1_exist_data_model.md [hunk 2]
```diff
@@ -354,7 +354,28 @@ Assume that you have already downloaded the checkpoints to the directory `checkp
...
+    ${OUT} \
+    [--exclude-extensions]
+```
+
+参数：
+
+- IMG_PATHS:  image root path.
+- CLASSES:  txt file name of storage class list.
+
+- OUT: output annotation json file name. The save dir is in the same directory as img_paths.
```
Comment (AronLin) [inline docs/1_exist_data_model.md:372]: The output annotation JSON file. The save directory is the same as IMG_PATHS.

File: tools/dataset_converters/images2coco.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import os
+
+import mmcv
+from PIL import Image
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Convert images to coco format without annotations')
+    parser.add_argument('img_paths', help='image root path')
```
Comment (RangiLyu) [inline tools/dataset_converters/images2coco.py:11]: img_paths or img_path

Comment (AronLin) [inline tools/dataset_converters/images2coco.py:11]: img_path, the root path of images

File: tools/dataset_converters/images2coco.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    parser.add_argument(
+        'classes', type=str, help='txt file name of storage class list')
```
Comment (AronLin) [inline tools/dataset_converters/images2coco.py:13]: same as above.

File: docs/1_exist_data_model.md [hunk 5]
```diff
@@ -354,7 +354,28 @@ Assume that you have already downloaded the checkpoints to the directory `checkp
...
+
+- exclude-extensions:  suffix name to be excluded.
```
Comment (AronLin) [inline docs/1_exist_data_model.md:374]: The suffix of images to be excluded, such as 'png' and 'bmp'.

File: docs/1_exist_data_model.md [hunk 6]
```diff
@@ -354,7 +354,24 @@ Assume that you have already downloaded the checkpoints to the directory `checkp
 
 ### Test without Ground Truth Annotations
 
-MMDetection supports to test models without ground-truth annotations using `CocoDataset`. If your dataset format is not in COCO format, please convert them to COCO format. For example, if your dataset format is VOC, you can directly convert it to COCO format by the [script in tools.](https://github.com/open-mmlab/mmdetection/tree/master/tools/dataset_converters/pascal_voc.py)
+MMDetection supports to test models without ground-truth annotations using `CocoDataset`. If your dataset format is not in COCO format, please convert them to COCO format. For example, if your dataset format is VOC, you can directly convert it to COCO format by the [script in tools.](https://github.com/open-mmlab/mmdetection/tree/master/tools/dataset_converters/pascal_voc.py) if your dataset format is Cityscapes, you can directly convert it to COCO format by the [script in tools.](https://github.com/open-mmlab/mmdetection/tree/master/tools/dataset_converters/cityscapes.py) The rest of the formats can be converted using [this script](https://github.com/open-mmlab/mmdetection/tree/master/tools/dataset_converters/images2coco.py).
```
Comment (AronLin) [inline docs/1_exist_data_model.md:357]: if your dataset format is Cityscapes,  if -> If.

**Review by AronLin — APPROVED** [review]
LGTM if the typo is fixed.

File: tools/dataset_converters/images2coco.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+def collect_image_infos(path, exclude_extensions=None):
+    img_infos = []
+
+    images_generator = mmcv.scandir(path, recursive=True)
+    for image_path in mmcv.track_iter_progress(list(images_generator)):
+        if exclude_extensions is None or (
+                exclude_extensions is not None
+                and not image_path.lower().endswith(exclude_extensions)):
+            image_path = os.path.join(path, image_path)
+            img_pillow = Image.open(image_path)
```
Comment (ZwwWayne) [inline tools/dataset_converters/images2coco.py:38]: suggest to use mmcv.imread which is faster

Comment (hhaAndroid) [reply tools/dataset_converters/images2coco.py:38]: `Image.open` just get the handle, not really read the picture. So it should be the fastest.

**Pull request merged**